cogs/guild_join.py

```diff
@@ -0,0 +1,80 @@
+import discord
+from discord.ext import commands
+
+from bot_globals import logger, STREAK_ROLES, MILESTONE_ROLES, VERIFIED_ROLE
+from embeds.misc_embeds import error_embed
+from embeds.stats_embeds import stats_embed, account_hidden_embed
+from embeds.users_embeds import account_not_found_embed
+from models.user_model import User
+from utils.middleware import admins_only, ensure_server_document, track_analytics
+
+class GuildJoin(commands.Cog):
+    def __init__(self, client):
+        self.client = client
+
+    # When the bot joins a server, create a new display information for each user in the server.
+    @commands.Cog.listener()
+    async def on_guild_join(self, guild) -> None:
+        print(f'Joined {guild.name} with {guild.member_count} members!')
+
+        # Create a new role
+        await generate_connected_users_role(self, guild)
+        await generate_milestone_users_role(self, guild)
+        await generate_streaks_users_role(self, guild)
+
+        return None
+    
+    @discord.app_commands.command(name="create-roles", description="Admins only: Creates roles on the server")
+    @ensure_server_document
+    @admins_only
+    @track_analytics
+    async def create_roles(self, interaction: discord.Interaction) -> None:
+        logger.info("file: cogs/guild_join.py ~ createRoles ~ run")
+
+        if not interaction.guild or not isinstance(interaction.channel, discord.TextChannel) or not isinstance(interaction.user, discord.Member):
+            embed = error_embed()
+            await interaction.response.send_message(embed=embed, ephemeral=True)
+            return
+        
+        await interaction.response.defer(ephemeral=True)
+        
+        # Create a new role
+        await generate_connected_users_role(self, interaction.guild)
+        await generate_milestone_users_role(self, interaction.guild)
+        await generate_streaks_users_role(self, interaction.guild)
+
+        await interaction.followup.send("Roles created successfully!", ephemeral=True)
```

Comment:
no need for ephemeral to be specified here as the ephemeral will be the same as the original deferred response